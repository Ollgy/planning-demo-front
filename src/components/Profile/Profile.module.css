.sectionContainer {
  height: 100%;
  padding-top: 20px;
  overflow-y: auto;
}

.formChangePswrd {
  position: relative;
}

.changePswrdBlock {
  position: absolute;
  top: 170px;
  right: 150px;
  background-color: #fff;
  padding: 20px;
  border: 2px solid #bfbfbf;
  border-radius: 15px;
  box-shadow: 5px 10px 10px 2px #e6e6e6, -5px 0px 10px 2px #e6e6e6;
}

.titleBlock {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  padding-right: 30px;
}

.formData {
  display: flex;
}

.formDataBlock {
}

.personalData {
  padding-top: 35px;
}

.personalData > *:last-of-type {
  margin-right: 0;
}

.photoBlock {
  width: 100%;
}

.infoBlock {
  width: 380px;
  margin-bottom: 15px;
}

.infoWrap {
  margin-right: 40px;
}

.changePswrd {
  margin-bottom: 25px;
}

.buttonBlock {
}

.closeButtonBlock {
  text-align: right;
  margin-bottom: 30px;
}

.msgBlock {
  width: 380px;
  height: 40px;
  margin-bottom: 15px;
}

.changePswrdLink {
  color: #f44f37;
}

/* transition */
.enter {
  display: block;
  opacity: 0.01;
}

.enter.enterActive {
  opacity: 1;
  transition: opacity 300ms, top 300ms ease-in;
}

.exit {
  opacity: 1;
}

.exit.exitActive {
  opacity: 0.01;
  transition: opacity 300ms;
}
