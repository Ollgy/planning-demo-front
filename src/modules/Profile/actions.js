import { createAction } from 'redux-actions';

export const showPswrdBlock = createAction('SHOW_PSWRD_BLOCK');
export const hidePswrdBlock = createAction('HIDE_PSWRD_BLOCK');
