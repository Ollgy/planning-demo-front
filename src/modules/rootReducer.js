import { combineReducers } from 'redux';
import authFromToken from './Auth';
import login from './Login';
import loginForm from './LoginForm';
import profile from './Profile';
import personalForm from './PersonalForm';
import changePswrdForm from './ChangePswrdForm';
import newTaskForm from './NewTaskForm';
import notes from './Notes';
import dashboard from './Dashboard';
import staff from './Staff';
import cleanBase from './CleanBase';

export default combineReducers({
  authFromToken,
  login,
  loginForm,
  profile,
  personalForm,
  changePswrdForm,
  newTaskForm,
  notes,
  dashboard,
  staff,
  cleanBase
});
