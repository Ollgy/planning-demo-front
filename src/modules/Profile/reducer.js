import { showPswrdBlock, hidePswrdBlock } from './actions';
import { combineReducers } from 'redux';
import { handleActions } from 'redux-actions';

const pswrdBlockOpen = handleActions(
  {
    [showPswrdBlock]: () => true,
    [hidePswrdBlock]: () => false
  },
  false
);

export default combineReducers({
  pswrdBlockOpen
});
