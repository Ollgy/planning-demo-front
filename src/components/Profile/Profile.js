import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { CSSTransition } from 'react-transition-group';
import SectionTitle from '../SectionTitle';
import ChangePswrdForm from '../ChangePswrdForm';
import PersonalForm from '../PersonalForm';
import { hidePswrdBlock, showPswrdBlock } from '../../modules/Profile';
import { M } from '../../const';
import styles from './Profile.module.css';

class Profile extends PureComponent {
  componentWillUnmount() {
    const { pswrdBlockOpen, hidePswrdBlock } = this.props;

    if (pswrdBlockOpen) {
      hidePswrdBlock();
    }
  }

  render() {
    const { user, pswrdBlockOpen, showPswrdBlock, hidePswrdBlock } = this.props;

    return (
      <section className={styles.sectionContainer}>
        <div className={styles.titleBlock}>
          <SectionTitle children="Мой профиль" size="m" />
          <div className={styles.changePswrd}>
            <a
              href=" "
              className={styles.changePswrdLink}
              onClick={e => {
                e.preventDefault();
                showPswrdBlock();
              }}
              children="Изменить пароль"
            />
          </div>
        </div>
        <div className={styles.personalData}>
          <div className={styles.formDataBlock}>
            <PersonalForm user={user} startMode={M.read} />
          </div>
          <CSSTransition
            in={pswrdBlockOpen}
            timeout={300}
            mountOnEnter={true}
            unmountOnExit={true}
            classNames={{
              appear: styles.appear,
              enter: styles.enter,
              enterActive: styles.enterActive,
              exit: styles.exit,
              exitActive: styles.exitActive
            }}
          >
            <div className={styles.changePswrdBlock}>
              <ChangePswrdForm user={user} hidePswrdBlock={hidePswrdBlock} />
            </div>
          </CSSTransition>
        </div>
      </section>
    );
  }
}

export default connect(
  state => ({ pswrdBlockOpen: state.profile.pswrdBlockOpen }),
  { hidePswrdBlock, showPswrdBlock }
)(Profile);
