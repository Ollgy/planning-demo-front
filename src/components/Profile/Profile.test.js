import React from 'react';
import Profile from './Profile';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { createStore, compose, applyMiddleware } from 'redux';
import rootReducer from '../../modules/rootReducer';
import thunk from 'redux-thunk';

const store = createStore(rootReducer, {}, compose(applyMiddleware(thunk)));

describe('<Profile/> UI component', () => {
  const shallowExpect = compose(
    expect,
    toJSON,
    shallow
  );

  it('Renders correctly', () => {
    shallowExpect(<Profile store={store} />).toMatchSnapshot();
  });

  const wrapper = shallow(<Profile store={store} />);
  const inner = wrapper.children().first();

  it('contains pswrdBlock is closed', () => {
    expect(inner.props().pswrdBlockOpen).toEqual(false);
  });
});
