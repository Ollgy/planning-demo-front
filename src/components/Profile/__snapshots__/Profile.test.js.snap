// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> UI component Renders correctly 1`] = `
<ContextProvider
  value={null}
>
  <Profile
    hidePswrdBlock={[Function]}
    pswrdBlockOpen={false}
    showPswrdBlock={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
